test/e2e/yurt/iot.go

```diff
@@ -0,0 +1,142 @@
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	iotv1alpha2 "github.com/openyurtio/openyurt/pkg/apis/iot/v1alpha2"
+	"github.com/openyurtio/openyurt/test/e2e/util"
+	ycfg "github.com/openyurtio/openyurt/test/e2e/yurtconfig"
+)
+
+func generateTestVersions() []string {
+	return []string{"levski", "jakarta", "kamakura", "ireland", "minnesota"}
+}
+
+var _ = Describe("OpenYurt IoT Test", func() {
+	var platformAdminName string
+
+	ctx := context.Background()
+	k8sClient := ycfg.YurtE2eCfg.RuntimeClient
+	timeout := 60 * time.Second
+	platformadminTimeout := 5 * time.Minute
+	testVersions := generateTestVersions()
+	namespaceName := "iot-test-namespace"
+	nodePoolName := util.NodePoolName
+
+	createNamespace := func() {
+		ns := corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		}
+		Eventually(
+			func() error {
+				return k8sClient.Delete(ctx, &ns, client.PropagationPolicy(metav1.DeletePropagationForeground))
+			}).WithTimeout(timeout).WithPolling(time.Millisecond * 500).Should(SatisfyAny(BeNil(), &util.NotFoundMatcher{}))
+		By("make sure the needed namespace are removed")
+
+		res := &corev1.Namespace{}
+		Eventually(
+			func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{
+					Name: namespaceName,
+				}, res)
+			}).WithTimeout(timeout).WithPolling(time.Millisecond * 500).Should(&util.NotFoundMatcher{})
+		Eventually(
+			func() error {
+				return k8sClient.Create(ctx, &ns)
+			}).WithTimeout(timeout).WithPolling(time.Millisecond * 300).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
+	}
+
+	createPlatformAdmin := func(version string) {
+		Eventually(func() error {
+			return k8sClient.Delete(ctx, &iotv1alpha2.PlatformAdmin{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      platformAdminName,
+					Namespace: namespaceName,
+				},
+			})
+		}).WithTimeout(timeout).WithPolling(500 * time.Millisecond).Should(SatisfyAny(BeNil(), &util.NotFoundMatcher{}))
+
+		testPlatformAdmin := iotv1alpha2.PlatformAdmin{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      platformAdminName,
+				Namespace: namespaceName,
+			},
+			Spec: iotv1alpha2.PlatformAdminSpec{
+				Version:  version,
+				PoolName: nodePoolName,
+			},
+		}
+		Eventually(func() error {
+			return k8sClient.Create(ctx, &testPlatformAdmin)
+		}).WithTimeout(timeout).WithPolling(500 * time.Millisecond).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
+	}
+
+	BeforeEach(func() {
+		By("Start to run iot test, clean up previous resources")
+		k8sClient = ycfg.YurtE2eCfg.RuntimeClient
+		createNamespace()
+	})
+
+	AfterEach(func() {
+		By("Cleanup resources after test")
+		By(fmt.Sprintf("Delete the entire namespaceName %s", namespaceName))
+		Expect(k8sClient.Delete(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}, client.PropagationPolicy(metav1.DeletePropagationBackground))).Should(BeNil())
+	})
+
+	for _, testVersion := range testVersions {
```

Comment:
The version here is edgex, and the platformadmin version is supposed to follow openyurt, so is it possible to create only one platformadmin in the e2e test, and leave the adaptation for the different edgex versions to the iot-dock e2e test?